### PR TITLE
[MKW] Ensure that a valid combination is being used

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by mkwcat** [comment]
Can you also add a check to make sure only standard karts / standard bikes are used in battle?

File: payload/import/mkwRegistry.hpp [hunk 1]
```diff
@@ -48,6 +128,40 @@ enum class Course {
...
+        return true;
+    default:
+        return false;
+    }
+}
+
+static constexpr bool IsVehicleValid(Vehicle vehicle)
+{
+    return vehicle >= Vehicle::StandardKartSmall && vehicle <= Vehicle::Phantom;
+}
```
Comment (vabold) [inline payload/import/mkwRegistry.hpp]: Check the gamemode in `IsCombinationValid` to determine which function to use for validation.

```suggestion
static constexpr bool IsVehicleValidVS(Vehicle vehicle)
{
    return vehicle >= Vehicle::StandardKartSmall && vehicle <= Vehicle::Phantom;
}

static constexpr bool IsVehicleValidBT(Vehicle vehicle)
{
    switch (vehicle) {
    case Vehicle::StandardKartSmall:
    case Vehicle::StandardKartMedium:
    case Vehicle::StandardKartLarge:
    case Vehicle::StandardBikeSmall:
    case Vehicle::StandardBikeMedium:
    case Vehicle::StandardBikeLarge:
        return true;
    default:
        return false;
    }
}
```

**Review by vabold — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
> Can you also add a check to make sure only standard karts / standard bikes are used in battle?

Good catch, addressed in commit [21bbff1](https://github.com/WiiLink24/wfc-patcher-wii/commit/21bbff10dc858823f734faf80d808bed8ce3eb23).

File: payload/import/mkwRegistry.hpp (hunk 1 above)
Comment (ghost) [reply payload/import/mkwRegistry.hpp]: Implemented in commit [21bbff1](https://github.com/WiiLink24/wfc-patcher-wii/commit/21bbff10dc858823f734faf80d808bed8ce3eb23).

**Pull request merged**